content/blog/model-calibration/index.Rmd

```diff
@@ -0,0 +1,411 @@
+
+There are two variables that control the windows. The **step size**, controls the frequency of the windows. If we set a step size of 5%, will create a new window every 5% probability (5%, 10%, 15%... etc). The second argument is the (maximum) **window size**. If it is set to 10%, then a given step will overlap halfway into the previous step, as well as the next step. Here is a visual representation of this specific scenario:
+
+```{r}
+#| echo: false
+#| fig.width: 6
+#| fig.height: 4
+#| out.width: 70%
+windows_size <- 0.10
+step_size    <- 0.05
+
+steps <- seq(0, 1, by = step_size)
+lower_cut <- steps - (windows_size / 2)
+lower_cut[lower_cut < 0] <- 0
+upper_cut <- steps + (windows_size / 2)
+upper_cut[upper_cut > 1] <- 1
+
+tibble(
+  step = steps,
+  lower = lower_cut,
+  upper = upper_cut,
+  group = seq_along(steps)
+) %>% 
+  ggplot(aes(x = step, y = group)) + 
+  geom_errorbar(aes(xmin = lower, xmax = upper)) +
+  labs(y = NULL, x = "predicted probability") +
+  theme_minimal()
+```
+
+In probably, the `cal_plot_windowed()` function provides this functionality. 
+The default step size is 0.05, and can be changed via the `step_size` argument.
+The default window size is 0.1, and can be changed via the `window_size` argument.
+
+```{r}
+segment_logistic %>% 
+  cal_plot_windowed(Class, .pred_good)
+```
+
+Here is an example of reducing the `step_size` from 0.05, to 0.02. There are
+more than double the windows:
+
+```{r}
+segment_logistic %>% 
+  cal_plot_windowed(Class, .pred_good, step_size = 0.02)
+```
+
+## Model-Based
+
+Another way to visualize the performance is to fit a classification model of the
+events against the estimated probabilities. This is helpful because it avoids the use of
+pre-determined groupings. Another difference, is that we are not plotting 
+midpoints of actual results, but rather predictions based on those results. 
+
+The `cal_plot_logistic()` provides this functionality. By default, it uses a 
+logistic regression. There are two possible methods for fitting: 
+
+spline model, provided by the `mgcv` package. The idea is to visualize
+a smooth line based on the predictions based on the `smooth` argument:
+
+* `smooth = TRUE` (the default) fits a generalized additive model using splines. This allows for more flexible model fits. 
+* `smooth = FALSE` uses an ordinary logistic regression model with linear terms for the predictor.
+
+As an example: 
+
+```{r}
+segment_logistic %>% 
+  cal_plot_logistic(Class, .pred_good)
+```
+
+The cooresponding `glm()` model produces:
+
+```{r}
+segment_logistic %>% 
+  cal_plot_logistic(Class, .pred_good, smooth = FALSE)
+```
+
+## Additional options and features
+
+### Intervals
+
+The confidence intervals are visualized using the gray ribbon. The default
+interval is 0.9, but can be changed using the `conf_level` argument.  
```

Comment:
why is the default `0.9`?

Reply:
I think this is something that @topepo requested, do you have a different preference or suggested default?

Reply:
I think I just noticed it because `0.95` is so ingrained in me 😅  If `0.9` is the common thing to do in this context then that should be the default.

Reply:
It's a convention that I ~decreed~ adopted at the beginning of tidymodels. The variance in estimates becomes much larger as we get closer to one and 90% confidence seems a good middle ground. A lot of Bayesian people do the equivalent of 80% confidence.

Reply:
Ah, thanks for the explanation! 🙌